akka-actor/src/main/scala/akka/actor/Scheduler.scala

```diff
@@ -543,3 +549,14 @@ object Cancellable {
     def isCancelled: Boolean = false
   }
 }
+
+object Scheduler {
+
+  /**
+   * If a `TaskRunOnClose` is used in `scheduleOnce` it will be run when the `Scheduler` is
+   * closed (`ActorSystem` shutdown). This is needed for the internal shutdown of dispatchers
+   * in Akka and is not intended to be used by end user applications, but it's public because
+   * a custom implementation of `Scheduler` must also implement this.
+   */
+  trait TaskRunOnClose extends Runnable
```

Comment:
`@NotForUserExtension` then?

Reply:
As we allow custom schedulers that might be too strong..